c/combinations/Makefile

```diff
@@ -0,0 +1,23 @@
+# This file is part of the SV-Benchmarks collection of verification tasks:
+# https://github.com/sosy-lab/sv-benchmarks
+#
+# SPDX-FileCopyrightText: 2015-2016 Daniel Liew <dan@su-root.co.uk>
+# SPDX-FileCopyrightText: 2015-2020 The SV-Benchmarks Community
+#
+# SPDX-License-Identifier: Apache-2.0
+
+LEVEL := ../
+
+CLANG_WARNINGS := \
+	-Wno-sometimes-uninitialized \
+	-Wno-uninitialized \
+
+include $(LEVEL)/Makefile.config
+
+license-pals:
+	reuse addheader --template header.jinja2 --copyright "The SV-Benchmarks Community" --year 2014-2020 --license Apache-2.0 pals_lcr.*c
```

Comment:
Preferably, we would not need to hard-code this here, because it might get outdated and incomplete.

What I meant in https://github.com/sosy-lab/sv-benchmarks/pull/1104#issuecomment-662830753 was to change `generate-tasks.py` such that it scans through the included C files and adds a union of all `SPDX-FileCopyrightText` lines to the output, and a line with `"SPDX-License-Identifier: " + " AND ".join(sorted(set(licenses)))` where `licenses` contains the licenses of each included C file. Is this feasible?

Reply:
Sorry, your comment was perfectly clear - I just had the slight hope that this easy solution is good enough.

> Is this feasible?

Yes, I'll see what I can do.

Reply:
SPDX copyright information is now properly extracted from the original tasks the new tasks are created from.

I've made some assumptions about the original copyright headers to be able to remove them from the combination tasks.
Namely, if the below condition is true, a line of the original program is not included in the combination task:

```python
        _is_singleline_comment(line) and (
            "This file is part of" in line
            or "sv-benchmarks" in line
            or "SPDX" in line
            or line.strip() == "//"
        ):
```

Even if the condition is not strict enough, this will only exclude comments, so I hope thats ok.

Reply:
Hm, not sure what you mean? Simply taking all (comment) lines with `SPDX-FileCopyrightText` should be enough for the copyright. In principle, all (comment) lines with `Copyright` would be safer, but for the purpose of this repo we can use the former pattern I think.

For licenses, I was under the impression that one must merge the licenses into a single line. However, the [FAQ](https://reuse.software/faq/#partial-license) says that the current way also seems to be ok, though I find it a little bit vague. Did you do this on purpose after reading the FAQ?

Reply:
We combine the two tasks by concatenating the existing tasks, so the existing copyright information of these would be duplicated.
With the above comment, I was talking about removing these duplicate lines.

I can change the license to AND if this is preferred, I don‘t have any preference.

The script is only intended for generating a very specific set of benchmark tasks, so I don’t try to generalize if this may complicate the script.

Reply:
Ah, now I understand. You are actually textually copying all lines including comments, and then it makes sense to exclude those lines that will be made redundant by the new copyright header. I had assumed that because of the necessary identifier renamings you certainly parse the files and dump them again, such that no comments appear at all (like it happens e.g., for minifiers) and I had written the comment before having a look at whether my assumption is correct. Sorry!

Regarding generality: sure! It was not my intention to suggest an unnecessary broad implementation.

Reply:
Perfect, thanks for taking the time to look into it :-)